src/ripple/app/misc/Validations.cpp

```diff
@@ -46,38 +47,34 @@ class ValidationsImp : public Validations
     Application& app_;
     std::mutex mutable mLock;
 
-    TaggedCache<uint256, ValidationSet> mValidations;
+    static constexpr std::chrono::seconds expireAge = std::chrono::seconds{600};
+    beast::aged_unordered_map<uint256, ValidationSet, Stopwatch::clock_type,
+        beast::uhash<>> mValidations;
     ValidationSet mCurrentValidations;
     std::vector<STValidation::pointer> mStaleValidations;
 
     bool mWriting;
     beast::Journal j_;
 
 private:
-    std::shared_ptr<ValidationSet> findCreateSet (uint256 const& ledgerHash)
+    ValidationSet * findCreateSet (uint256 const& ledgerHash)
     {
-        auto j = mValidations.fetch (ledgerHash);
-
-        if (!j)
-        {
-            j = std::make_shared<ValidationSet> ();
-            mValidations.canonicalize (ledgerHash, j);
-        }
-
-        return j;
+        return &mValidations[ledgerHash];
     }
 
-    std::shared_ptr<ValidationSet> findSet (uint256 const& ledgerHash)
+    ValidationSet * findSet (uint256 const& ledgerHash)
```

Comment:
Correct, this was temporary while I refactored.  In the new version of the code in `Validations`, the equivalent lookup into the aged map still returns a `ValidationSet *`, but no public methods can return that set so the lifetime is guaranteed.

As you might be thinking, that is only "enforced" by convention and a soon-to-be-added comment.  An alternative I considered was to completely restrict access to the map and only allow iteration over it via a functor.